tools/dkcm/clerk/clerk.go

```diff
@@ -0,0 +1,112 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package clerk
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"knative.dev/test-infra/pkg/mysql"
+)
+
+// Cluster stores a row in the "Cluster" db table
+// Table Schema: knative.dev/test-infra/tools/dkcm/clerk/schema.sql
+type Cluster struct {
+	ClusterId   string
+	accessToken string
+	BoskosId    string
+	ProwId      string
+	Status      string
+	Zone        string
+	lastUpdate  time.Time
+}
+
+// ClusterConfig is the struct to return to Prow once the cluster is available
+type ClusterConfig struct {
+	ClusterName string
+	BoskosId    string
+	Zone        string
+}
+
+type ClerkOperations interface {
+	//check cluster available, if available, return cluster id and access token
+	CheckAvail() (bool, string, string)
+	// get with token
+	GetCluster(token string) (ClusterConfig, error)
+	// delete a cluster entry
+	DeleteCluster(accessToken string) error
+	// Insert a cluster entry
+	InsertCluster(accessToken string, boskosId string, prowId string, status string, zone string, lastUpdate time.Time) error
+	// List clutsers within a time interval
+	ListClusters(window time.Duration) ([]Cluster, error)
+	// Generate Unique access token for Prow
+	GenerateToken() string
+}
+
+// DB holds an active database connection. This implements all the fuunctions of ClerkOperations.
+type DB struct {
+	*sql.DB
+}
+
+func (c Cluster) String() string {
+	return ""
+}
+
+// NewDB returns the DB object with an active database connection
+func NewDB(c *mysql.DBConfig) (*DB, error) {
+	return nil, nil
+}
+
+func (db *DB) CheckAvail() (bool, string, string) {
+	return true, "", ""
+}
+
+// insert a cluster entry into db
+func (db *DB) InsertCluster(accessToken string, boskosId string, prowId string, status string, zone string) error {
+	return nil
+}
+
+// List all clusters within a certain time window
+func (db *DB) ListClusters(window time.Duration) ([]Cluster, error) {
+	var result []Cluster
+	return result, nil
+
+}
+
+// DeleteCluster deletes a row from Cluster db
+func (db *DB) DeleteCluster(accessToken string) error {
+	return nil
+}
+
+func (db *DB) GenerateToken() string {
+	return ""
+}
+
+func (db *DB) getCluster(token string) (*ClusterConfig, error) {
+	// check in with token
+	return &ClusterConfig{}, nil
+}
+
+// checkRowAffected number of checks row affected by a given query.
+func checkRowAffected(stmt *sql.Stmt, args ...interface{}) error {
+	return nil
+}
+
+// name cluster in the following format: e2e-cluster{id}-{prowid}
+func nameCluster(clusterId string, prowId string) string {
```

Comment:
This should be `prowJobID`, also note in Go, we prefer `ID` over `Id` - https://stackoverflow.com/a/39830270/13578870

Reply:
Thanks. Fixed